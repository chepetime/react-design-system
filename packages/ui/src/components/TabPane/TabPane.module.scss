@use "./../../styles" as styles;

:root {
  --tabpane-background-color: #{styles.$pink-500};
}

:global .dark {
  --tabpane-background-color: #{styles.$indigo-500};
}

.TabPane {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--tabpane-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
