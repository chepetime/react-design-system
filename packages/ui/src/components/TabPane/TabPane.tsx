import React from "react";
import classnames from "classnames";
import css from "./TabPane.module.scss";

export interface TabPaneProps extends React.ComponentPropsWithRef<"div"> {
  children?: React.ReactNode;
}

export const TabPane = React.forwardRef<HTMLDivElement, TabPaneProps>(
  (props: TabPaneProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.TabPane)} ref={ref}>
        <p>TabPane component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

TabPane.displayName = "TabPane";
