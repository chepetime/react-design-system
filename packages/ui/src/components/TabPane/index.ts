export { TabPane } from "./TabPane";
export type { TabPaneProps } from "./TabPane";