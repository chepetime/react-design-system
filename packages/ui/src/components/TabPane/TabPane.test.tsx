
import React from "react";
import userEvent from "@testing-library/user-event";
import "@testing-library/jest-dom";
import { TabPane } from "./TabPane";
import { render, screen } from "@testing-library/react";

describe("TabPane", () => {
  it("renders without errors", () => {
    render(<TabPane />);
    // Assert that the component renders without throwing any errors
  });

  it("renders with children", () => {
    render(<TabPane>Some children</TabPane>);
    // Assert that the component renders the provided children
    const childrenElement = screen.getByText("Some children");
    expect(childrenElement).toBeInTheDocument();
  });

  it("handles user interactions", () => {
    render(<TabPane>Some children</TabPane>);
    // Simulate user interactions with the component
    userEvent.click(screen.getByText("Some children"));
    // Assert the expected outcome of the user interaction
    // For example, check if a certain state has changed or an event handler has been called
  });

  it("matches snapshot", () => {
    const { container } = render(<TabPane />);
    // Assert that the rendered component matches a previously captured snapshot
    expect(container).toMatchSnapshot();
  });
});
