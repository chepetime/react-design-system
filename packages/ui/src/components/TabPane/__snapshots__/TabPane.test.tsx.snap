// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPane matches snapshot 1`] = `
<div>
  <div
    class="TabPane"
  >
    <p>
      TabPane component working!
    </p>
    <div />
  </div>
</div>
`;
